src/SnD.Sdk/Storage/Amazon/AmazonSqsService.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Threading.Tasks;
+using Akka;
+using Akka.Streams.Dsl;
+using Akka.Streams.SQS;
+using Amazon.SQS;
+using Amazon.SQS.Model;
+using Microsoft.Extensions.Logging;
+using Snd.Sdk.Storage.Base;
+using Snd.Sdk.Storage.Models;
+using Snd.Sdk.Tasks;
+
+namespace Snd.Sdk.Storage.Amazon;
+
+/// <summary>
+/// Queue Service implementation for AWS SQS.
+/// </summary>
+public class AmazonSqsService : IQueueService<AmazonSqsSendResponse, AmazonSqsReleaseResponse>
+{
+    private readonly IAmazonSQS client;
+    private readonly ILogger<AmazonSqsService> logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AmazonSqsService"/> class.
+    /// </summary>
+    /// <param name="client">The Amazon SQS client.</param>
+    /// <param name="logger">The logger instance.</param>
+    public AmazonSqsService(IAmazonSQS client, ILogger<AmazonSqsService> logger)
+    {
+        this.client = client;
+        this.logger = logger;
+    }
+
+    private string GetQueueUrlAsync(string queueName)
+    {
+        return this.client.GetQueueUrlAsync(new GetQueueUrlRequest
+        {
+            QueueName = queueName
+        }).GetAwaiter().GetResult().QueueUrl;
+    }
+
+    /// <inheritdoc />
+    public Task<AmazonSqsSendResponse> SendQueueMessage(string queueName, string messageText)
+    {
+        this.logger.LogDebug("Sending {messageText} to {queueName}", messageText, queueName);
+
+        var messageRequest = new SendMessageRequest()
+        {
+            QueueUrl = GetQueueUrlAsync(queueName),
+            MessageBody = messageText
+        };
+        return this.client.SendMessageAsync(messageRequest).Map(result => new AmazonSqsSendResponse
+        { MessageId = result.MessageId, SequenceNumber = result.SequenceNumber });
+    }
+
+    /// <inheritdoc />
+    public Source<QueueElement, NotUsed> GetQueueMessages(string queueName, TimeSpan visibilityTimeout,
+        int prefetchCount, TimeSpan pollInterval)
+    {
+        this.logger.LogDebug("Creating a stream from queue: {queueName}, using visibility timeout {visibilityTimeout}",
+            queueName, visibilityTimeout);
+        var settings = SqsSourceSettings.Default
+            .WithVisibilityTimeout(visibilityTimeout)
+            .WithMaxBatchSize(Math.Min(10, prefetchCount))
```

Comment:
Why do you set a minimum value here?

Reply:
so I avoid having a value larger than 10, 10 being the maximum allowed